### ISSUE TITLE
Propagate save errors up to the user

### ISSUE THREAD

**Issue by dysbulic** [ceramicstudio/self.id#30]
I was having difficulty saving my profile. On the site all that showed was a "failed to save" error. In the console, there was only a `POST` `500` error with no description.

Only by going to the network debugging tab and clicking on the failed request was I able to see the validation error: "`data/residenceCountry must NOT have more than 2 characters, data/residenceCountry must match pattern \"^[A-Z]{2}$\"`".

I think validation errors should propagate up to the user and, if possible, be associated with the appropriate form field.

Also, the editing interface shouldn't close until it has successfully saved or the user hits cancel. As it is, everything has to be reentered when it fails to save.

**Comment by oed** [comment]
I believe the  `residenceCountry` must be a country code, e.g. *SE*.
cc @PaulLeCam

**Comment by dysbulic** [comment]
> I believe the `residenceCountry` must be a country code, e.g. _SE_.

Yeah, the requirement makes sense. It's failing without any indication as to why that I was saying should be addressed.

**Comment by scottrepreneur** [comment]
> `residenceCountry` validation
>  \"^[A-Z]{2}$\"

Please add this to some docs on IDX! cc: @PaulLeCam  🙏

**Comment by PaulLeCam** [comment]
> > `residenceCountry` validation
> > "^[A-Z]{2}$"
> 
> Please add this to some docs on IDX! cc: @PaulLeCam 🙏

It is specified in the CIP, see https://github.com/ceramicnetwork/CIP/blob/main/CIPs/CIP-19/CIP-19.md#record-schema
